src/logic/note.ts

```diff
@@ -0,0 +1,206 @@
+import { useLiveQuery } from "dexie-react-hooks";
+import { CardType } from "./card";
+import { db } from "./db";
+import { v4 as uuidv4 } from "uuid";
+import { Deck } from "./deck";
+import { Table } from "dexie";
+
+export interface NoteSkeleton {
+  id: string;
+  deck: string;
+  referencedBy: string[];
+  creationDate: Date;
+  customOrder?: number;
+}
+
+export interface Note<T extends CardType> extends NoteSkeleton {
+  content: NoteContent<T>;
+}
+
+export function createNoteSkeleton(deck: string): NoteSkeleton {
+  return {
+    deck: deck,
+    creationDate: new Date(Date.now()),
+    id: uuidv4(),
+    referencedBy: [],
```

Comment:
wouldn't be "easier" to just fetch cards using the "note" field? with a query?

Reply:
Thanks!

implemented it, should work

I think there might be some other things that could be simplified by taking more advantage of Dexies's features...